workflows/sarek.nf

```diff
@@ -1001,6 +1008,64 @@ workflow SAREK {
             rt_file
         )
 
+        if (params.concatenate_vcfs) {
+            // Concatenate vcf-files
+
+            // Gather vcfs and vcf-tbis for concatenating germline-vcfs
+            germline_vcfs_with_tbis = Channel.empty()
+
+            germline_vcfs_with_tbis = germline_vcfs_with_tbis.mix(
+                BAM_VARIANT_CALLING_GERMLINE_ALL.out.deepvariant_vcf.join(BAM_VARIANT_CALLING_GERMLINE_ALL.out.deepvariant_vcf_tbi)
+            )
+
+            germline_vcfs_with_tbis = germline_vcfs_with_tbis.mix(
+                BAM_VARIANT_CALLING_GERMLINE_ALL.out.freebayes_vcf.join(BAM_VARIANT_CALLING_GERMLINE_ALL.out.freebayes_vcf_tbi)
+            )
+
+            germline_vcfs_with_tbis = germline_vcfs_with_tbis.mix(
+                BAM_VARIANT_CALLING_GERMLINE_ALL.out.haplotypecaller_vcf.join(BAM_VARIANT_CALLING_GERMLINE_ALL.out.haplotypecaller_vcf_tbi)
+            )
+
+            germline_vcfs_with_tbis = germline_vcfs_with_tbis.mix(
+                BAM_VARIANT_CALLING_GERMLINE_ALL.out.manta_vcf.join(BAM_VARIANT_CALLING_GERMLINE_ALL.out.manta_vcf_tbi)
+            )
+
+            germline_vcfs_with_tbis = germline_vcfs_with_tbis.mix(
+                BAM_VARIANT_CALLING_GERMLINE_ALL.out.mpileup_vcf.join(BAM_VARIANT_CALLING_GERMLINE_ALL.out.mpileup_vcf_tbi)
+            )
+
+            germline_vcfs_with_tbis = germline_vcfs_with_tbis.mix(
+                BAM_VARIANT_CALLING_GERMLINE_ALL.out.strelka_vcf.join(BAM_VARIANT_CALLING_GERMLINE_ALL.out.strelka_vcf_tbi)
+            )
+
+            germline_vcfs_with_tbis = germline_vcfs_with_tbis.mix(
+                BAM_VARIANT_CALLING_GERMLINE_ALL.out.tiddit_vcf.join(BAM_VARIANT_CALLING_GERMLINE_ALL.out.tiddit_vcf_tbi)
+            )
+
+            germline_vcfs_with_tbis = germline_vcfs_with_tbis.map{
+                meta, vcf, tbi ->
+                    def new_meta = meta.clone()
```

Comment:
the meta.clone doesn't do a deep copy. we'll need to copy all the fields. However, the remove might actually work as is. IS there a reason to clone this on every step?

Reply:
The many clone-statements might just be me being to eager with the copy-paste 😆  I'll try to get rid of them, and then test.